msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CreateBindingResourcePackageBase.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Xml;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.MacDev.Tasks {
+	public abstract class CreateBindingResourcePackageBase : Task {
+		public string SessionId { get; set; }
+		
+		[Required]
+		public string OutputPath { get; set; }
+		
+		[Required]		
+		public ITaskItem[] NativeReferences { get; set; }
+		
+		[Required]
+		public string ProjectDir { get; set; }
+		
+		[Required]
+		public string BindingAssembly { get; set; }
+		
+		public override bool Execute ()
+		{
+			// LinkWith must be migrated for NoBindingEmbedding styled binding projects
+			if (NativeReferences.Length == 0)
+				Log.LogError ($"Unable to create a Binding Resource Package with no Native References. Move from LinkWith.cs to NativeReferences or disable NoBindingEmbedding");
+
+			string bindingResourcePath = Path.Combine (ProjectDir, OutputPath, $"{Path.GetFileNameWithoutExtension (BindingAssembly)}.resources");
+			Log.LogMessage ($"Creating binding resource package: {bindingResourcePath}");
+
+			Directory.CreateDirectory (bindingResourcePath);
+			foreach (var nativeRef in NativeReferences)
+				CopyNativeReference (nativeRef.ItemSpec, bindingResourcePath);
+
+			CreateManifest (bindingResourcePath);
+
+			return true;
+		}
+
+		// There is no built-in C# API for recursive copy of directories
+		void CopyNativeReference (string nativeRef, string resourcePath)
+		{
+			// TODO - Should this be smarter if we didn't change? Can't use smart copy unless framework? rsync?
+			// TODO - Is this safe for Windows builds?
+			var copyTask = new Microsoft.Build.Tasks.Exec {
```

Comment:
Can you differentiate non-windows builds so that you can use the new, hot file cloning on OS X?

Reply:
there's already a `CopyTask` that handle all that right ?

Reply:
It will take some more lines but yea better use [Copy Task](https://docs.microsoft.com/en-us/visualstudio/msbuild/copy-task?view=vs-2017)